src/app/myworkflows/sidebar-accordion/github-apps-logs/lambda-events-datasource.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *  Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import { CollectionViewer, DataSource } from '@angular/cdk/collections';
+import { HttpResponse } from '@angular/common/http';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { finalize } from 'rxjs/operators';
+import { LambdaEvent, LambdaEventsService } from '../../../shared/openapi';
+
+export type ShowContent = 'table' | 'error' | 'empty' | 'noResult' | null;
+
+export class LambdaEventDataSource implements DataSource<LambdaEvent> {
+  private eventsSubject$ = new BehaviorSubject<LambdaEvent[]>([]);
+  private loadingSubject$ = new BehaviorSubject<boolean>(true);
+  private eventsLengthSubject$ = new BehaviorSubject<number>(0);
+  private showContentSubject$ = new BehaviorSubject<ShowContent>(null);
+
+  public events$ = this.eventsSubject$.asObservable();
+  public loading$ = this.loadingSubject$.asObservable();
+  public eventsLength$ = this.eventsLengthSubject$.asObservable();
+  public showContent$ = this.showContentSubject$.asObservable();
```

Comment:
Is the reason for this Subject->Observable mapping to prevent the Subjects from being publically exposed as "writable" objects?  A Subject is an Observable: https://rxjs.dev/guide/subject

Reply:
> Is the reason for this Subject->Observable mapping to prevent the Subjects from being publically exposed as "writable" objects? A Subject is an Observable: https://rxjs.dev/guide/subject

That was my thinking, yes. I didn't want to give outside consumers of the class the ability to push to feed the Observable.